lnrpc/marshall_utils.go

```diff
@@ -179,3 +180,57 @@ func MarshalOutPoint(op *wire.OutPoint) *OutPoint {
 		OutputIndex: op.Index,
 	}
 }
+
+// Parse the coin selection strategy string from the command line
+// to an equivalent lnrpc.CoinSelectionStrategy enum type.
+func ParseCoinSelectionStrategyString(strategy string)(
+	*CoinSelectionStrategy, error) {
+
+	var coinSelectionStrategy CoinSelectionStrategy
+	switch strategy {
+	case "global-config":
+		coinSelectionStrategy = CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG
+
+	case "largest":
+		coinSelectionStrategy = CoinSelectionStrategy_STRATEGY_LARGEST
+
+	case "random":
+		coinSelectionStrategy = CoinSelectionStrategy_STRATEGY_RANDOM
+
+	default:
+		return nil, fmt.Errorf("unknown coin selection strategy "+
+			"%v", strategy)
+	}
+
+	return &coinSelectionStrategy, nil
+}
+
+// Parse the coins selection strategy lnrpc.CoinSelectionStrategy enum type
+// to an equivalent wallet.CoinSelectionStrategy type.
+func ParseCoinSelectionStrategyEnum(strategy CoinSelectionStrategy)(
+	wallet.CoinSelectionStrategy, error) {
+
+	var coinSelectionStrategy wallet.CoinSelectionStrategy
+
+	switch strategy {
+	case CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG:
+		strategyValue, _ := ParseCoinSelectionStrategyString(
+			"largest",
+		)
+		coinSelectionStrategy, _ = ParseCoinSelectionStrategyEnum(
+			*strategyValue,
+		)
+
+	case CoinSelectionStrategy_STRATEGY_LARGEST:
+		coinSelectionStrategy = wallet.CoinSelectionLargest
```

Comment:
nit: can return directly in the `case` statements, then no final `return` at the end of the function is needed (since there's a `default` case).